common/python/utility/hex_utils.py

```diff
@@ -29,6 +29,10 @@ def hex_to_utf8(binary):
     return binascii.hexlify(binary).decode("UTF-8")
 
 
+def hex_to_byte_array(hex_str):
```

Comment:
Add a docstring.  Adapted from hex_string.h:

```
     """
     Convert a hex string (i.e., a string of characters with values      
     between '0'-'9', 'A'-'F') to an array of bytes.
     """
```

Reply:
Add UTs for the function. Functions in util should be exhaustively tested as it is exposed to all other modules.

Reply:
Done